### PR TITLE
feat(useAttachments): use asset id order instead of settings

### PR DESCRIPTION
**Pull request by fulopdaniel**
Updated app bridge version because I received the following error for each package during typecheck:

Argument of type 'AppBridgeBlock' is not assignable to parameter of type 'AppBridgeBlock | AppBridgeTheme'.

### CODE REVIEW

File: packages/shared/src/hooks/useAttachments.ts [hunk 1]
```diff
@@ -1,47 +1,44 @@
...
-    const [blockSettings, setBlockSettings] = useBlockSettings(appBridge);
-    const { blockAssets, updateAssetIdsFromKey, addAssetIdsToKey, deleteAssetIdsFromKey } = useBlockAssets(appBridge);
+export const useAttachments = (appBridge: AppBridgeBlock, assetSettingsKey: string) => {
+    const { blockAssets, updateAssetIdsFromKey } = useBlockAssets(appBridge);
     const attachments = blockAssets?.[assetSettingsKey] || [];
-    const attachmentsInBlockSettings = (blockSettings[blockSettingsKey] || []) as { id: number }[];
 
-    const onAddAttachments = async (assets: Asset[]) => {
+    const onAddAttachments = async (newAssets: Asset[]) => {
         const newAssetIds = [];
```
Comment (ragi96) [inline packages/shared/src/hooks/useAttachments.ts:10]:
```suggestion
        const existingAssetId = attachments.map((attachment) => attachment.id);
```

File: packages/shared/src/hooks/useAttachments.ts [hunk 2]
```diff
@@ -1,47 +1,44 @@
...
-        for (const asset of assets) {
+
+        for (const attachment of attachments) {
+            newAssetIds.push(attachment.id);
+        }
```
Comment (ragi96) [inline packages/shared/src/hooks/useAttachments.ts:14]: Can be removed if you change the initialisation

File: packages/shared/src/hooks/useAttachments.ts [hunk 3]
```diff
@@ -1,47 +1,44 @@
...
+        for (const asset of newAssets) {
             newAssetIds.push(asset.id);
         }
```
Comment (ragi96) [inline packages/shared/src/hooks/useAttachments.ts:17]:
```suggestion
        newAssetIds.push(...newAssets.map((asset) => asset.id));
```

File: packages/shared/src/hooks/useAttachments.ts [hunk 4]
```diff
@@ -1,47 +1,44 @@
...
-        await setBlockSettings({
-            [blockSettingsKey]: attachmentsInBlockSettings.filter(({ id }) => id !== assetToDelete.id),
-        });
-        await deleteAssetIdsFromKey(assetSettingsKey, [assetToDelete.id]);
+        const newAssetIds = [];
+        for (const attachment of attachments) {
+            if (attachment.id !== assetToDelete.id) {
+                newAssetIds.push(attachment.id);
+            }
+        }
```
Comment (ragi96) [inline packages/shared/src/hooks/useAttachments.ts:27]:
```suggestion
    const newAssetIds = attachments
        .filter((attachment) => attachment.id !== assetToDelete.id)
        .map((attachment) => attachment.id);
```

File: packages/shared/src/hooks/useAttachments.ts [hunk 5]
```diff
@@ -50,20 +47,15 @@ export const useAttachments = (appBridge: AppBridgeBlock, blockSettingsKey: stri
         for (const asset of assets) {
             newAssetIds.push(asset.id);
         }
```
Comment (ragi96) [inline packages/shared/src/hooks/useAttachments.ts:49]: could be something like that:

```suggestion
        const newAssetIds = assets.map((asset) => asset.id);
```

File: packages/shared/src/hooks/useAttachments.ts [hunk 6]
```diff
@@ -1,47 +1,44 @@
...
-        });
         const newAssetIds = [];
+
         for (const attachment of attachments) {
             if (attachment.id !== attachmentToReplace.id) {
                 newAssetIds.push(attachment.id);
+            } else {
+                newAssetIds.push(newAsset.id);
             }
         }
```
Comment (ragi96) [inline packages/shared/src/hooks/useAttachments.ts:41]: Could be something like that:

```suggestion
         const newAssetIds = attachments.map((attachment) => {
            if (attachment.id !== attachmentToReplace.id) {
                return attachment.id;
            } else {
                return newAsset.id;
            }
        });
```

**Review by ragi96 — COMMENTED** [review]
Update also the app-bridge within the examples

File: packages/shared/src/hooks/useAttachments.ts (hunk 6 above)
Comment (fulopdaniel) [reply packages/shared/src/hooks/useAttachments.ts]: Good catch. As discussed, for when the assets array comes from the asset chooser it has a custom iterable which does not support map. For the other occasions I updated the code.

**Review by ragi96 — APPROVED** [review]
🚀

File: packages/shared/src/hooks/useAttachments.ts [hunk 7]
```diff
@@ -1,69 +1,46 @@
...
+        await updateAssetIdsFromKey(assetSettingsKey, newAssetIds);
+    };
 
     return {
         onAddAttachments,
         onAttachmentDelete,
         onAttachmentReplace,
         onAttachmentsSorted,
-        sortedAttachments,
+        sortedAttachments: attachments,
```
Comment (SamuelAlev) [inline packages/shared/src/hooks/useAttachments.ts:44]:
```suggestion
       attachments,
```

File: packages/audio-block/src/components/BlockAttachments.tsx [hunk 8]
```diff
@@ -11,7 +11,7 @@ export const BlockAttachments = ({ appBridge }: BlockAttachmentsProps) => {
         onAttachmentDelete,
         onAttachmentReplace,
         onAttachmentsSorted,
-    } = useAttachments(appBridge, ATTACHMENTS_SETTING_ID, ATTACHMENTS_SETTING_ID);
+    } = useAttachments(appBridge, ATTACHMENTS_SETTING_ID);
```
Comment (SamuelAlev) [inline packages/audio-block/src/components/BlockAttachments.tsx:14]:
```suggestion
    } = useAttachments(appBridge, ATTACHMENTS_ASSET_ID);
```

It isn't a setting ID but an asset ID

File: packages/image-block/src/components/Image.tsx [hunk 9]
```diff
@@ -65,7 +65,7 @@ export const ImageComponent = ({ image, appBridge, blockSettings }: ImageProps)
         onAttachmentDelete,
         onAttachmentReplace,
         onAttachmentsSorted,
-    } = useAttachments(appBridge, ATTACHMENTS_SETTING_ID, ATTACHMENTS_SETTING_ID);
+    } = useAttachments(appBridge, ATTACHMENTS_SETTING_ID);
```
Comment (SamuelAlev) [inline packages/image-block/src/components/Image.tsx:68]:
```suggestion
    } = useAttachments(appBridge, ATTACHMENTS_ASSET_ID);
```

File: packages/image-block/src/components/Image.tsx
Comment (SamuelAlev) [inline packages/image-block/src/components/Image.tsx:1]: You can also remove the `attachments: { id: number }[];` in `types.ts`

**Review by Walkeezy — APPROVED** [review]

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**